js/unit_tests.ts

```diff
@@ -107,6 +108,35 @@ test(function tests_readFileSync_NotFound() {
 });
 */
 
+/* TODO(ry) Add this once we can create a tmpDir to write the file into.
+test(function writeFileSyncSuccess() {
+  const enc = new TextEncoder();
+  const dataWritten = enc.encode("Hello");
+  const filename = "TEMPDIR/test.txt";
+  deno.writeFileSync(filename, dataWritten, 0o666);
+  const dataRead = readFileSync(filename);
+  assertEqual(dataRead, dataWritten);
+});
+*/
+
+// For this test to pass we need --allow-write permission.
+// Otherwise it will fail with deno.PermissionDenied instead of deno.NotFound.
+test(function writeFileSyncFail() {
+  const enc = new TextEncoder();
+  const data = enc.encode("Hello");
+  const filename = "/baddir/test.txt";
+  // The following should fail because /baddir doesn't exist (hopefully).
+  let caughtError = false;
+  try {
+    deno.writeFileSync(filename, data);
+  } catch (e) {
+    caughtError = true;
+    // TODO assertEqual(e, deno.NotFound);
```

Comment:
I don't understand. You want to have a "const" error object for all NotFound error objects?
I'd rather have see relevant information (such as the failing path) as properties on the object.